src/visitors/global_var_visitor.cpp

```diff
@@ -0,0 +1,77 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <iostream>
+#include <memory>
+
+#include "ast/global.hpp"
+#include "ast/neuron_block.hpp"
+#include "ast/program.hpp"
+#include "ast/range.hpp"
+#include "ast/statement_block.hpp"
+#include "visitors/global_var_visitor.hpp"
+
+namespace nmodl {
+namespace visitor {
+
+void GlobalToRangeVisitor::visit_neuron_block(ast::NeuronBlock& node) {
+    ast::RangeVarVector range_variables;
+    std::vector<std::string> global_variables_to_remove;
+    std::vector<ast::StatementVector::const_iterator> global_statements_to_remove;
+
+    auto& statement_block = node.get_statement_block();
+    auto& statements = (*statement_block).get_statements();
+    const auto& symbol_table = ast->get_symbol_table();
+
+    for (auto statements_it = statements.begin(); statements_it < statements.end();
```

Comment:
I think this is a copy. Isn't it better const auto&

Reply:
Just to be fair with Ioannis :smiley: I asked him to remove some [extra ones here](https://github.com/BlueBrain/nmodl/pull/230#discussion_r429069086).